fdbserver/storageserver.actor.cpp

```diff
@@ -5453,6 +5616,33 @@ ACTOR Future<Void> updateStorage(StorageServer* data) {
 		state Version desiredVersion = data->desiredOldestVersion.get();
 		state int64_t bytesLeft = SERVER_KNOBS->STORAGE_COMMIT_BYTES;
 
+		// Clean up stale checkpoint requests, this is not supposed to happen, since checkpoints are cleaned up onup
+		// failures. This is kept as a safeguard.
+		while (!data->pendingCheckpoints.empty() && data->pendingCheckpoints.begin()->first <= startOldestVersion) {
+			for (int idx = 0; idx < data->pendingCheckpoints.begin()->second.size(); ++idx) {
+				auto& metaData = data->pendingCheckpoints.begin()->second[idx];
+				data->actors.add(deleteCheckpointQ(data, startOldestVersion, metaData));
+				TraceEvent("StorageStaleCheckpointRequest", data->thisServerID)
```

Comment:
Since this is unexpected, consider logging at Severity=30.

Reply:
Done. Thanks for the suggestion.